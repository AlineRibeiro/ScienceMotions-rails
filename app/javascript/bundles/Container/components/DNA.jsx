import React from "react";
import Adenine from "images/adenine.png";
import Cytosine from "images/cytosine.png";
import Guanine from "images/guanine.png";
import Thymine from "images/thymine.png";

const DNA = () => (
  <section id="visual">
    <div className="table">
      <div className="row">
        <div className="col-sm-1" />
        <div className="col-sm-3">
          <img src={Cytosine} className="dna-img c" />
          <img src={Thymine} className="dna-img t" />
          <img src={Adenine} className="dna-img a" />
          <img src={Guanine} className="dna-img g" />
        </div>
      </div>
    </div>
  </section>
);

export default DNA;
