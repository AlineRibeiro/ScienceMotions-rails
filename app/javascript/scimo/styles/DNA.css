.dna-img {
  position: fixed;
  width: 50%;
  height: 50%;
}

.dna-img.t {
  top: 0;
  left: 0;
}

.dna-img.c {
  top: 0;
  right: 0;
}

.dna-img.a {
  bottom: 0;
  left: 0;
}

.dna-img.g {
  bottom: 0;
  right: 0;
}

section#top {
  position: relative;
  width: 100%;
  height: 750px;
  overflow: hidden;
  text-align: center;
}
