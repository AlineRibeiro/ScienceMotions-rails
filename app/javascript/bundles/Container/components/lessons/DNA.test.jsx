import React from 'react';
import renderer from 'react-test-renderer';

import DNA from './DNA';
describe('LeaveCourseForm', () => {
  it('renders correctly when there are no items', () => {
    const rendered = renderer.create(<DNA />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
})
