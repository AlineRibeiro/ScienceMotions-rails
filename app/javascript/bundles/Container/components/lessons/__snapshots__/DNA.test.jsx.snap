// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaveCourseForm renders correctly when there are no items 1`] = `
Array [
  <div
    id="skrollr-body"
  >
    <div
      data-0="background-image: linear-gradient(90deg, hsl(172, 100%, 50%), hsl(318, 68%, 68%)"
      data-450="background-image: linear-gradient(450deg, hsl(172, 100%, 50%), hsl(318, 68%, 68%)"
    >
      <section
        id="visual"
      >
        <img
          className="dna-img t"
          src="test-file-stub"
        />
        <img
          className="dna-img c"
          src="test-file-stub"
        />
        <img
          className="dna-img a"
          src="test-file-stub"
        />
        <img
          className="dna-img g"
          src="test-file-stub"
        />
      </section>
    </div>
  </div>,
  <section
    id="text"
  >
    <div
      className="table"
    >
      <h1>
        Structure of DNA
      </h1>
      <p>
        Each gene in a molecule of DNA contains:
      </p>
      <p>
        _A different sequence of
        <span
          id="text-1"
        >
           bases
        </span>
      </p>
      <p>
        _Codes for a particular protein
      </p>
      <p>
        In DNA, two strands coil together to form a double helix. There are chemical cross-links between the two strands, formed by
        <span
          id="text-2"
        >
           pairs of bases
        </span>
      </p>
      .
      <p>
        Each strand of DNA is made of chemicals called bases. Do not confuse these with the bases you meet when you study acids and alkalis in chemistry. There are four different types of bases, shown as
        <span
          id="text-3"
        >
           A, T, C and G
        </span>
        .
      </p>
      <p>
        Proteins are made in the cytoplasm of a cell, not in the nucleus. Genes cannot leave the nucleus, so a copy of the gene is needed. This copy is able to leave the nucleus to go into the cytoplasm so that proteins can be made by the cell.
      </p>
    </div>
  </section>,
]
`;
